patches/tModLoader/Terraria/ModLoader/GlobalNPC.cs

```diff
@@ -92,6 +92,10 @@ public virtual void ModifyHoverBoundingBox(NPC npc, ref Rectangle boundingBox)
 	{
 	}
 
+	public virtual void OnHoverBoundingBox(NPC npc, Rectangle boundingBox, bool mouseIntersects, bool smartCursorInteraction)
```

Comment:
1. Needs docs
2. `smartCursorInteraction` isn't quite the right name for this variable